mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -773,17 +793,28 @@ impl Nexus {
     }
 
     /// returns the current status of the nexus
-    pub fn status(&self) -> NexusState {
-        self.state
-    }
-
-    pub async fn get_rebuild_progress(
-        &self,
-    ) -> Result<RebuildProgressReply, Error> {
-        // TODO: add real implementation
-        Ok(RebuildProgressReply {
-            progress: "Not implemented".to_string(),
-        })
+    pub fn status(&self) -> NexusStatus {
+        match self.state {
+            NexusState::Init => NexusStatus::Degraded,
+            NexusState::Closed => NexusStatus::Faulted,
+            NexusState::Open => {
+                if self
+                    .children
+                    .iter()
+                    .all(|c| c.status() == ChildStatus::Online)
+                {
+                    NexusStatus::Online
+                } else if self
+                    .children
+                    .iter()
+                    .any(|c| c.status() == ChildStatus::Online)
```

Comment:
These else/if statements seem to be the same? Also, can you comment the reasoning why (right now) Closed gets mapped to degraded?

Reply:
I think Closed gets mapped to faulted -> line 799
or are you talking about the children closed? The idea was that (I'm starting to comment the code as well):
/// Status of the nexus
    /// Online
    /// All children must also be online
    /// Degraded
    /// At least one child must be online
    /// Faulted
    /// No child is online so the nexus is faulted

Reply:
If im not reading outdated code again, what it said was:

if .iter().... any(|c| Online) else if { .iter()...any(|c| Online }

So both branches were evaluating the same expression namely "is the child online"

Reply:
I'm not sure, the code is actually if .iter().... **all**(|c| Online) else if { .iter()...**any**(|c| Online }